server/src/main/java/org/opensearch/identity/authz/PERMISSIONS_MODEL.md

```diff
@@ -0,0 +1,52 @@
+
+To further clarify this change consider the difference between the two lenses of the scenario outlined on the [OpenSearch Permission Documentation](https://opensearch.org/docs/latest/security-plugin/access-control/permissions/):
+
+Imagine that a subject `User_1` seeks to perform a `_bulk` request using the following:
+
+```
+POST _bulk
+{ "delete": { "_index": "test-index", "_id": "tt2229499" } }
+{ "index": { "_index": "test-index", "_id": "tt1979320" } }
+{ "title": "Rush", "year": 2013 }
+{ "create": { "_index": "test-index", "_id": "tt1392214" } }
+{ "title": "Prisoners", "year": 2013 }
+{ "update": { "_index": "test-index", "_id": "tt0816711" } }
+{ "doc" : { "title": "World War Z" } }
+```
+
+Under the previous Permissions model, `User_1` needed to have four separate permissions for `test_index` which all required verification each time the operation would be performed.
+
+Under the new model of Permissions, it is possible to simply check the Identity of `User_1` for the required permissions and then after authorizing the request once, the operation can be assumed authorized until the authorization token expires. This Permission model is similar to the
+authentication model introduced with the Identity changes and makes use of a Permission model that maintains a least privileges model by assuming all Subjects are permission-less upon initial authentication.
+After a Subject is authenticated, Permission tokens are only added to their Identity upon the Subjects first attempt to operate using them. Returning to the example, when `User_1` logs into OpenSearch, they are assumed to have no permissions.
+Then, when they attempt to perform the `_bulk` request, the appropriate Permissions checks are executed for each of the operations constituting their request. As each operation is approved, a decaying token is added to their Identity which signifies
+that for the life of the token, any further attempts to perform the same type of operation are automatically authorized. After a decay period, the token is reset and the subject will once again require authorization in order to perform the requests.
+Tokens are also reset everytime the authentication token of the Subject is reset.
+
+
+### Permission Nomenclature
+
+With the restructuring of the Permissions Model, a simplification of the Permissions Nomenclature is in order. Instead of the previous naming convention which
+emphasized the different operations which required permissions at the `cluster` and `index` level, the new naming convention takes an Identity-focused structure.
+
+For any given Permission, the terminology of the specific operation can be broken into a constituent pattern `<Subject>.<Scope>.<Operation>.<Token_Hash>`. For instance,
+returning to the above `_bulk` request example, the Permission associated with `User_1` deleting data from `test-index` would be: `User_1.test-index.delete.<Token_Hash>`.
```

Comment:
nit: line-break not needed

Reply:
> @scrawfor99 thanks for the PR. I love the new changes and moving towards identity.
> 
> Sorry if these items are already covered in the Identity doc (I haven't read it yet).
> 
>     * What is the AuthZ Tenant (client) workflow and its interaction with AuthZ Server.
> 
>     * Who is the AuthZ server?
> 
>     * Do we invalidate an issued token if the user Identity policy changes?
> 
>     * Usual web auth's add an API to refresh the token, do we plan on to add one?
> 
>     * How do we plan on to support backwards compatibility? I am worried we have lots of customers with existing AuthZ model and we should start thinking about how to migrate them smoothly.

Hi @saratvemulapalli, thank you for taking the time to reach out.

To address each of your questions, I am going to paraphrase from the Identity Roadmap document. 

In short, the current authorization mechanism is being reworked to align with an identity-focused version where authentication and authorization are being handled as a part of core. Right now, we have aspects of the authentication and authorization logic handled in the Security plugin as well as the OpenSearch core server library. We are likely going to be migrating most if not all of the logic from the plugin model into core and then combining it with parts of the server code to create an Identity Library containing both authentication and authorization modules. The concept of tenancy is something that we are planning on revisiting.

The basic plan for the authorization tokens is to make similar use to an authentication token where they have short-lived expiration times. These tokens will grant a Subject the ability to complete a pre-approved (already-authorized) operation during their life but will expire both at the end of their time to live as well as on any changes in the Subject authentication. This means that if you were to have your authentication token expire your authorization token would also expire. 

Currently, the plan was to implement a refreshing mechanism similar to the authentication token setup. 

For backward compatibility, the shift to an Identity-based model is first goign to start as an optional Sandbox version of OpenSearch. The greater details of the migration are still being discussed but the changes will be part of a breaking-change major release. 

I am tagging @peternied who outlined much of the Identity work so may be a better person to answer some of your questions.